lib/types/duration.js

```diff
@@ -127,6 +121,8 @@ Duration.prototype.toString = function () {
  * @returns {Duration}
  */
 Duration.fromBuffer = function (buffer) {
+  if (!buffer instanceof Buffer)
+    throw new TypeError('Expected Buffer type');
```

Comment:
Is the goal of this an earlier incompatible type detection?

Reply:
Sort of. But as this function is unchanged from original, it may make more sense to don't do it.